File: AoC-2024/Day-01/2024-01-1.js
// https://www.mitchellmudd.dev/blog/read-text-file-nodejs
// https://www.w3schools.com/js/js_string_methods.asp
// https://www.w3schools.com/jsref/jsref_split.asp

const fs = require('fs')
txt = fs.readFileSync('Input.txt', 'utf8')

l_0 = []
l_1 = []

lines = txt.split('\n')
for (i = 0; i < lines.length; i++) {
    a = lines[i].split(' ')   // Get the ith line, then split on spaces
    l_0[i] = Number(a[0])
    l_1[i] = Number(a[a.length-1])
}

l_0.sort()
l_1.sort()

ans = 0
for (i = 0; i < l_0.length; i++) {
    ans += Math.abs(l_0[i]-l_1[i])
}
console.log(ans)



// Recycling Bin ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

// fs.readFile('Sample.txt', 'utf8', (err, data) => {
//     if (err) {
//       console.error(err);
//       return;
//     }
//     console.log(data);
// });

// fetch('Sample.txt').then(r=>r.text()).then(t=>console.log(t));

File: AoC-2024/Day-01/2024-01-1.py
with open('Input.txt', 'r') as f:
    txt = f.read()

l_0,l_1 = [],[]
for line in txt.split('\n'):
    line = [int(i) for i in line.split()]
    l_0.append(line[0])
    l_1.append(line[1])

l_0.sort()
l_1.sort()

ans = 0
for i in range(len(l_0)):
    ans += abs(l_1[i]-l_0[i])

print(ans)
